### DISCUSSION TITLE
parent folder of git repos

### DISCUSSION THREAD

**Discussion by mircealungu** [git-truck/git-truck#522] [Q&A]
How complicated would it be to open the parent folder of a series of git repos? 

I envision a way of opening the folder with the 10 student projects in my course, and see all of them at once. 
Also we could see the relative sizes of the projects.

I also envision opening all my python projects and seeing their relative size and activity.
Would also help a company that has a mono-repository: see the state of all their projects. 

And now that we have the zooming-in... it would be so easy to focus on a single project too. 

Probably future work... but it seems to me that it would be quite cool.

**Comment by nimrossum** [comment]
I think it is doable.

- It would require running the analyzer for each sub repo and merging them into a single tree structure. 
- We would need to iterate over each sub folder and check that they are git repositories and "ignore" the ones that aren't. 
- We would have to default to the checked out branch in the repository.

**Comment by nimrossum** [comment]
Performancewise, we need to maybe disable animations and only show names for `x` largest folders.

**Comment by mircealungu** [comment]
The performance optimization you suggest is important also for large individual repos.

Alternatively, for a large repo it might make sense to automatically start with the treemap view? assuming that bending the labels is what slows the renderer?

**Comment by mircealungu** [comment]
or maybe disable labels during animations?

**Comment by nimrossum** [comment]
@mircealungu 
> Would also help a company that has a mono-repository: see the state of all their projects.

I think we should already support monorepositories, unless you mean something different than a repository with a lot of sub projects in them. Are you thinking of repositories with [Submodules?](https://git-scm.com/book/en/v2/Git-Tools-Submodules)

**Comment by mircealungu** [comment]
Mono-repositories are indeed supported. I don't know what I meant with that :)

**Comment by nimrossum** [comment]
@mircealungu We have made it easier to open multiple repositories. We have not implemented visualizing multiple repositories at the same time.

**Comment by mircealungu** [comment]
Nice! Looking forward to seeing how this works!
At the same time, pls. send me the latest version of the survey. 
I'd like to do another pass over it!
Then, we have to plan the marketing: reddit, linkedin, facebook, twitter, own connections, ... we should get as much data as possible ;)

M.

**Comment by nimrossum** [comment]
@mircealungu 
Oh, we will for sure! We are working on getting one last feature merged.

You can take a look at the survey here:
https://docs.google.com/forms/d/1LBmjl88XrIAY-SHlHU1jUMSbxsnkmNef5NcctXJacQM/edit?usp=drivesdk&chromeless=1
Note that some of the stock photos will be removed.

**Comment by mircealungu** [comment]
at a first glance

- amazing! i never new you can have such colorful surveys! 

- file extension is not a “metric” - please call those “views”

- "And then try to see if you are able to setup and use Git Truck on your computer”
  can we give them a better defined task than “installing it on their computer”?

- "You can attach a screenshot here”
  like it’d be their most burning wish, and we just let them know how to do it?

- project that you analyzed … 
  do we want to push them towards a certain type of project? maybe one that’s most collaborative? or at least that has a lot of history (for the churn views?)

M.

**Comment by nimrossum** [comment]
@mircealungu 

> - file extension is not a “metric” - please call those “views”

Yeah that makes sense, changed it.

> - "And then try to see if you are able to setup and use Git Truck on your computer”
>   can we give them a better defined task than “installing it on their computer”?

We tried to keep it pretty vague, to test how easy it is to get started and whether the README makes sense, what do you think about that?

> - "You can attach a screenshot here”
>   like it’d be their most burning wish, and we just let them know how to do it?

Maybe we could instruct them how to do it, or provide a button i the UI to download a screenshot?


> - project that you analyzed …
>   do we want to push them towards a certain type of project? maybe one that’s most collaborative? or at least that has a lot of history (for the churn views?)

I think that is a good idea!

**Comment by mircealungu** [comment]
> 
> "And then try to see if you are able to setup and use Git Truck on your computer”
> can we give them a better defined task than “installing it on their computer”?
> We tried to keep it pretty vague, to test how easy it is to get started and whether the README makes sense, what do you think about that?
> 
sure. 

as far as installations go, your tool is the easiest
i’ve seen! and it’s targeted at programmers :) so you
should not waste too much time on this is my intuition.

> "You can attach a screenshot here”
> like it’d be their most burning wish, and we just let them know how to do it?
> Maybe we could instruct them how to do it, or provide a button i the UI to download a screenshot
> 
I meant, explain to them that we would appreciate if they could provide
a screenshot. it’s extra work. so explain why should they bother. 

> project that you analyzed …
> do we want to push them towards a certain type of project? maybe one that’s most collaborative? or at least that has a lot of history (for the churn views?)
> I think that is a good idea!
> 
cool.

m.

**Reply by nimrossum** [reply]
@mircealungu 
We have made some edits, if you don't any more comments, feel free to forward the questionnaire to your peers!
